### PR TITLE
FWI-2476: Add missingNetworkPolicy, automountServiceAccountToken, and linuxHardening checks

### PR DESCRIPTION
**Pull request by ivanfetch-fw**
## Checklist
* [x] I have signed the CLA
* [x] I have updated/added any relevant documentation

### What's the goal of this PR?
Add checks for

* missingNetworkPolicy - requires a NetworkPolicy that contains applied egress and ingress rules, and targets a Pod label.
* automountServiceAccountToken - requires that auto mounting of a ServiceAccount token is disabled for a Pod, either at the ServiceAccount or Pod level.
* linuxHardening - requires that one of seLinux options, AppArmor, sec comp profile, or dropping Linux capabilities are used to prevent a container from using unwanted privileges. Some of these can be set at the Pod or Container level.
### What changes did you make?
Added the above checks along with success and failure schema test-cases to validate fields which can be set or overridden at the pod and container spec.

Internal Ticket: [FWI-2476](https://fairwinds.myjetbrains.com/youtrack/issue/FWI-2476)

### CODE REVIEW

**Review by makoscafee — APPROVED** [review]

File: checks/linuxHardening.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
+successMessage: One of AppArmor, Seccomp, SELinux, or dropping Linux Capabilities are used to restrict containers using unwanted privileges
+FailureMessage: Use one of AppArmor, Seccomp, SELinux, or dropping Linux Capabilities to restrict containers using unwanted privileges
+category: Security
+target: Container
```
Comment (makoscafee) [inline checks/linuxHardening.yaml:4]: Here the target is `Container` but you are checking `SecompProfile` from PodSpec.SecurityContext and not checking for `Containers.SecurityContext.` was this on purpose?

File: checks/missingNetworkPolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        labels:
+          type: object
+          minProperties: 1
+additionalSchemaStrings:
+  networking.k8s.io/NetworkPolicy: |
+    type: object
+    properties:
+      spec:
+        type: object
+        required: ["podSelector", "egress", "ingress"]
```
Comment (makoscafee) [inline checks/missingNetworkPolicy.yaml:21]: egress and ingress are optional any one of these could be present. do we want to enforce the presence of both?

**Review by makoscafee — APPROVED** [review]
LGTM, just had few questions .

File: checks/linuxHardening.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          {{ end }}
+          properties:
+            seccompProfile:
+              type: object
+              required:
+              {{ if or (not $podSeccompProfileType) (eq $podSeccompProfileType "Unconfined") }}
+                - type
+              {{ end }}
+              properties:
+                type:
```
Comment (ivanfetch-fw) [inline checks/linuxHardening.yaml:30]: @makoscafee RE: your comment
> Here the target is Container but you are checking SecompProfile from PodSpec.SecurityContext and not checking for Containers.SecurityContext. was this on purpose?

Here is where the per-container schema requires `seccompProfile.type` depending on what the pod-level field is set to; if the pod-level field is empty or set to `Unconfined` the container field must be set to something other than `Unconfined`.

File: checks/missingNetworkPolicy.yaml (hunk 2 above)
Comment (ivanfetch-fw) [reply checks/missingNetworkPolicy.yaml:21]: In this case yes - the ticket associated with this check requires both.

File: checks/linuxHardening.yaml (hunk 1 above)
Comment (ivanfetch-fw) [reply checks/linuxHardening.yaml:4]: The per-container schema is requiring `securityContext` depending on the pod-level field. I commented with more detail in line 30 of this file.

**Comment by ivanfetch-fw** [comment]
Barnie [confirmed he doesn't have additional comments and this can be merged](https://fairwindsops.slack.com/archives/CNQBGRGFN/p1659714043413909?thread_ts=1658858841.999019&cid=CNQBGRGFN).

**Pull request merged**